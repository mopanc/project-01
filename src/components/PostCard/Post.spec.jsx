import { render, screen } from '@testing-library/react';
import { PostCard } from '.';
import { PostCardPropsMock } from './mock';


const props = PostCardPropsMock;

describe('<PostCard />', () => {
  it('should render PostCard correctly', () => {
    render(<PostCard {...props} />)

    expect(screen.getByRole('img', { name: /Title 1/i })).toHaveAttribute('src', 'img/img.png');
    expect(screen.getByRole('heading', { name: /title 1/i })).toBeInTheDocument();
    expect(screen.getByText('Body 1')).toBeInTheDocument();
  });

  it('should mach snapshot', () => {
    const { container } = render(<PostCard {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  })
});