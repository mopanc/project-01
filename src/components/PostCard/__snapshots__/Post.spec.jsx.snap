// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should mach snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Title 1"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h1>
      Title 1
       
    </h1>
    <p>
      Body 1
    </p>
    <span>
      1
    </span>
  </div>
</div>
`;
